### PR TITLE
add useraccount crud endpoints

### PR DESCRIPTION
**Pull request by CarlosQ96**
Added:
1. UserAccounts GET by AccountId, PUT by accountId, and POST to create ACcount Id

Caveats:
1. Who generated the accountId? 
2. Is the accountId the same as the userId (looking at the discord bot).
3. Should a new migration to add the new event log type be done? or is the original one rerun? How are we handling backward improvement on the new api? (it will work on new deploys but old ones it might not)

### CODE REVIEW

**Comment by CarlosQ96** [comment]
You can start taking a look Kris! I have a few question I left in the description.
I think this base works, but I think some modifications might be done, as I might have a few confusions after looking at the discord bot.

**Comment by CarlosQ96** [comment]
Also the linter threw many errors, on all files, and running the linter locally also does errors everywhere. 
linter --fix does not work.

Any suggestions for linting or how are you handling it?

**Review by kristoferlund — CHANGES_REQUESTED** [review]
Nice @CarlosQ96 !!

The different ids in play here are: 

`user._id` – MongoDb User Id

`useraccount._id` – MongoDb User Id
`useraccount.accountId` – Platform specific account id, in the case of Discord it is the Discord user snowflake id 

So, for your questions: 

1. Discord in our current case. When adding Slack, that will have another accountId.
2. No, see above.
3. Yes, create a new migration, old ones are not rerun.

I realized we need to be able to GET and PATCH both with `user._id` and `useraccount.accountId`. I don't know how to best structure the endpoints. 

Can we use same endpoints and allow both types of ids? Mongoose provides this easy check: `Types.ObjectId.isValid(id)`

**Comment by kristoferlund** [comment]
> Also the linter threw many errors, on all files, and running the linter locally also does errors everywhere. linter --fix does not work.
> 
> Any suggestions for linting or how are you handling it?

For linting, I get this: `✖ 37 problems (3 errors, 34 warnings)`. Do you see more? I'll open up a new issue to fix these.

File: packages/api2/src/useraccounts/useraccounts.service.ts [hunk 1]
```diff
@@ -42,6 +92,8 @@ export class UserAccountsService {
   ): Promise<UserAccount | null> {
     const userAccount = await this.userAccountModel
       .findOne({ userAccountId })
+      .select('_id user accountId name avatarId platform createdAt updatedAt')
```
[the remark this answers is not in the document]
Comment (CarlosQ96) [reply packages/api2/src/useraccounts/useraccounts.service.ts:95]: Thanks will do

**Comment by CarlosQ96** [comment]
> > Also the linter threw many errors, on all files, and running the linter locally also does errors everywhere. linter --fix does not work.
> > Any suggestions for linting or how are you handling it?
> 
> For linting, I get this: `✖ 37 problems (3 errors, 34 warnings)`. Do you see more? I'll open up a new issue to fix these.

I get 38 problems, I think its the same.

**Comment by CarlosQ96** [comment]
Thanks for the feedback will add it today, I knew we had to do changes as I was unsure about a few things. But we can reuse this base to finish it.
Regarding the Ids, I look into making it restful. Or use the same ones, maybe something more elegant.

**Comment by kristoferlund** [comment]
```
GET /useraccount?id=xxx&accountId=xxx
PATCH /useraccount?id=xxx&accountId=xxx
```

**Comment by CarlosQ96** [comment]
Fixed the duck db issue I had, made the changed just missing to fix 1 test. I will push it first thing tomorrow morning. (my morning)

**Pull request merged**